lib/internal/bootstrap/loaders.js

```diff
@@ -289,26 +285,27 @@ NativeModule.prototype.proxifyExports = function() {
 };
 
 NativeModule.prototype.compile = function() {
-  const id = this.id;
+  if (this.loaded || this.loading) {
+    return this.exports;
+  }
 
+  const id = this.id;
   this.loading = true;
 
   try {
     const requireFn = this.id.startsWith('internal/deps/') ?
-      NativeModule.requireWithFallbackInDeps :
-      NativeModule.require;
+      requireWithFallbackInDeps : nativeModuleRequire;
 
     const fn = compileFunction(id);
     fn(this.exports, requireFn, this, process, internalBinding, primordials);
 
-    if (experimentalModules && this.canBeRequiredByUsers) {
```

Comment:
Note that since we no longer hits `canBeRequiredByUsers` for the internal `require`, we should be able to remove the immediate dependency `exposeInternals` in `loaders.js` and get the value from C++ later when necessary - only after bootstrap is complete, of course.